src/gentropy/dataset/study_locus.py

```diff
@@ -574,6 +574,41 @@ def clump(self: StudyLocus) -> StudyLocus:
         )
         return self
 
+    def exclude_region(
+        self: StudyLocus, region: str, exclude_overlap: bool = False
+    ) -> StudyLocus:
+        """Exclude a region from the StudyLocus dataset.
+
+        Args:
+            region (str): region given in "chr##:#####-####" format
+            exclude_overlap (bool): If True, excludes StudyLocus windows with any overlap with the region.
```

Comment:
> exclude_overlap (bool): If True, excludes StudyLocus windows with any overlap with the region.

I am not 100% sure if this is what happening here. You are excluding all study loci where the lead snp falls within the locus. Not if the start and end position overlap. If I understood it right.

Reply:
That's right. By default (exclude_overlap=False) it just looks at the position column, so the position of the lead variant for that StudyLocus. When True, it instead removes any rows where the StudyLocus' window overlaps with the region given to the function.

Reply:
So it doesn't care if the actual locusStart / locusEnd overlaps with the filtered region. OK.